app/customsqlhandler.py

```diff
@@ -0,0 +1,296 @@
+      if unique_name_no_period in all_input:
+        if all_input.get(unique_name_no_period) != '':
+          field.previous_value = all_input.get(unique_name_no_period)
+
+      if unique_name_no_period in data:
+        field.previous_value = data.get(unique_name_no_period)
+
+    all_input.update(data)
+    session['custom_sql_previous_filters'] = all_input
+    session['custom_sql_previous_checked_fields'] = checked_boxes
+
+    all_records = get_customsql_records(all_input)
+
+    all_records_json = all_records.json()
+
+    search_criteria = {}
+    search_range, records_per_page = page_bounds(page_number)
+    search_criteria.update(search_range)
+    get_microservice_info_json = all_records_json
+
+    if len(get_microservice_info_json) > 0:
+      microservice_sum = get_microservice_info_json[0].get('total_records', 0)
+      max_page = math.ceil(microservice_sum / records_per_page)
+    else:
+      microservice_sum = 0
+      max_page = 1
+
+    table_records = get_table_records(
+        field_classes,
+        get_microservice_info_json,
+        custom_sql=True,
+    )
+
+    result_message_str = result_message(search_range, microservice_sum,
+                                        "Custom SQL")
+
+    table_headers = get_table_headers(field_classes)
+
+    views, current_view_index = get_views(user_role, 'customsql')
+    header_html = get_html(user_role, views)
+    current_view = views[current_view_index]
+
+    return {
+        'views': views,
+        'header_html': header_html,
+        'current_view': current_view,
+        'fields': fields,
+        'database_names': database_names,
+        'called_from_index': 'customsql',
+        'Fields': field_classes,
+        'result_message': result_message_str,
+        'page_title': f'Custom SQL view for: {user_role}',
+        'page_number': page_number,
+        'last_page_number': max_page,
+        'table_headers': table_headers,
+        'table_records': table_records,
+    }
+
+  @aiohttp_jinja2.template('microservices.html')
+  async def get(self, request):
+    session = await get_session(request)
+    data = await request.post()
+
+    user_role = await saml.get_role_id(request)
+
+    await saml.ensure_logged_in(request)
+
+    clear = request.match_info['clear']
+    if clear == 'clear':
+      temp = session.get('custom_sql_previous_filters', {})
+      final = {}
+      for key in temp:
+        final[key] = ''
+      session['custom_sql_previous_filters'] = final
+
+    if microservices_permissions(user_role, 'customsql') == False:
+      request['client_ip'] = request.get('client_ip', "No IP Provided")
+      return await forbidden(request)
+
+    database_names, fields = await get_database_fields(request)
+    page_number = get_page(request)
+
+    checked_boxes = session.get('custom_sql_previous_checked_fields', [])
+
+    client_input = {}
+    all_input = {}
+    field_classes = []
+    error = ""
```

Comment:
Single quotes! Also not sure this needs to be initialized here.

Reply:
Remved